### ISSUE TITLE
Admission controller is being created before the namespace

### ISSUE THREAD

**Issue by ArjonBu** [argoproj/argo-cd#4854]
**Describe the bug**

I am trying to deploy an environment using Kustomize and Argocd. 

I have noticed that Argocd will try to create the admission controller objects before actually creating the namespace. I can see in the UI that the admission controller objects are in the "pre-sync" phrase. This is not done by me via annotations but it's done by Argocd.

I tried to add the "pre-sync" flag to the namespace object, but this makes argocd remove and redeploy this namespace for every change. This is not a solution. 

**To Reproduce**

Have a Kustomize base with this manifest https://github.com/kubernetes/ingress-nginx/blob/master/deploy/static/provider/cloud/deploy.yaml

**Expected behavior**

I would expect Argocd to create the namespace and then deploy objects in this namespace.

**Screenshots**

**Version**

```shell
argocd: v1.7.8+ef5010c.dirty
  BuildDate: 2020-10-16T04:56:45Z
  GitCommit: ef5010c3a0b5e027fd642732d03c5b0391b1e574
  GitTreeState: dirty
  GoVersion: go1.15.2
  Compiler: gc
  Platform: darwin/amd64
argocd-server: v1.7.8+ef5010c
  BuildDate: 2020-10-15T22:34:12Z
  GitCommit: ef5010c3a0b5e027fd642732d03c5b0391b1e574
  GitTreeState: clean
  GoVersion: go1.14.1
  Compiler: gc
  Platform: linux/amd64
  Ksonnet Version: v0.13.1
  Kustomize Version: {Version:kustomize/v3.6.1 GitCommit:c97fa946d576eb6ed559f17f2ac43b3b5a8d5dbd BuildDate:2020-05-27T20:47:35Z GoOs:linux GoArch:amd64}
  Helm Version: version.BuildInfo{Version:"v3.2.0", GitCommit:"e11b7ce3b12db2941e90399e874513fbd24bcb71", GitTreeState:"clean", GoVersion:"go1.13.10"}
  Kubectl Version: v1.17.8
```

**Comment by jannfis** [comment]
This behaviour has been fixed w/ https://github.com/argoproj/gitops-engine/pull/159, which will be released with the upcoming v1.8

**Comment by ArjonBu** [comment]
Closing this issue then. Thanks
